### PR TITLE
Wrapper storage package

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
## Description of the reasoning behind the pull request
- Refactoring dependencies to elrond-go-storage repo
  
## Proposed Changes
 - Move all changes to a single storage adapter package
-  Move back to elrond-go the path manager and directory handler

### CODE REVIEW

File: storage/storageunit/constants.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package storageunit
+
+import "github.com/ElrondNetwork/elrond-go-storage/storageUnit"
+
+const (
+	LRUCache     = storageUnit.LRUCache
```
Comment (iulianpascalau) [inline storage/storageunit/constants.go:6]: comments on exported items?

File: storage/errors.go [hunk 2]
```diff
@@ -77,6 +70,30 @@ var ErrNilNodeTypeProvider = errors.New("nil node type provider")
 // ErrNilOldDataCleanerProvider signals that a nil old data cleaner provider has been provided
 var ErrNilOldDataCleanerProvider = errors.New("nil old data cleaner provider")
 
+// ErrNilCacher is raised when a nil cacher is provided
```
Comment (sstanculeanu) [inline storage/errors.go:73]: not used anywhere

File: storage/pathmanager/errors.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+// ErrInvalidStaticPathTemplate signals that an invalid path template for static storers has been provided
+var ErrInvalidStaticPathTemplate = errors.New("invalid path template for static storers")
+
+// ErrInvalidDatabasePath signals that an invalid database path has been provided
+var ErrInvalidDatabasePath = errors.New("invalid database path")
+
+// IsNotFoundInStorageErr returns whether an error is a "not found in storage" error.
+// Currently, "item not found" storage errors are untyped (thus not distinguishable from others). E.g. see "pruningStorer.go".
+// As a workaround, we test the error message for a match.
+func IsNotFoundInStorageErr(err error) bool {
```
Comment (sstanculeanu) [inline storage/pathmanager/errors.go:26]: not used

File: storage/constants.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package storage
+
+import (
+	"time"
+)
+
+// MaxRetriesToCreateDB represents the maximum number of times to try to create DB if it failed
+const MaxRetriesToCreateDB = 10
```
Comment (sstanculeanu) [inline storage/constants.go:8]: as this is used only in the `storage` package and it also exists in `elrond-go-storage`, why not use the one from `elrond-go-storage` repo? this way we avoid missing updating one of them in one of the repos

same for `SleepTimeBetweenCreateDBRetries`

Comment (AdoAdoAdo) [reply storage/constants.go:8]: the constants from elrond-go-storage were removed:
https://github.com/ElrondNetwork/elrond-go-storage/pull/4/files

**Review by ssd04 — COMMENTED** [review]
- it looks like some of the interfaces in `storage/interface.go` are not being used like `SerializedStoredData`, `LRUCacheHandler`, `ForEachItem`, maybe we can remove them
- also others like Cacher seems to be duplicated with storage repo, maybe use an alias?

File: dataRetriever/txpool/memorytests/memory_test.go [hunk 5]
```diff
@@ -229,7 +229,11 @@ func pprofHeap(scenario *scenario, step string) {
 
 	defer func() {
 		errClose := file.Close()
-		panic(fmt.Sprintf("cannot close file: %s", errClose.Error()))
+		errMsg := "nil"
+		if errClose != nil {
+			errMsg = errClose.Error()
+		}
+		panic(fmt.Sprintf("cannot close file: %s", errMsg))
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply dataRetriever/txpool/memorytests/memory_test.go:236]: done

File: storage/storageunit/constants.go (hunk 1 above)
Comment (AdoAdoAdo) [reply storage/storageunit/constants.go:6]: done

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**